lib/pure/complex.nim

```diff
@@ -376,6 +376,15 @@ proc `$`*(z: Complex): string =
 
 {.pop.}
 
+proc FFT*(dataSource : seq[float]):seq[Complex]= 
```

Comment:
Thanks for your comments, this is my first formal contribution to an open source project .  AFAIK, fft  is commonly used in AI procedure to reflect time-domain signal to frequency-domain to abstract regular pattern which is not discoverable in the former.It will be handy if Nim provides a build-in fft. @dom96,Where should I put the document?  As for algorithm, this is a translation from part of our patent codes, it seems Huawei was cheated :) ,at least it is correct , @jcosborn , what name do you suggest? Complex is minority except for science computing, I think it belongs to complex module for now.